packages/gestalt/src/TextField/InternalTextField.js

```diff
@@ -149,10 +151,12 @@ const InternalTextFieldWithForwardRef: AbstractComponent<Props, HTMLInputElement
     disabled ? formElement.disabled : formElement.enabled,
     (hasError || hasErrorMessage) && !focused ? formElement.errored : formElement.normal,
     {
-      [layout.medium]: !tags && size === 'md',
-      [layout.large]: tags || size === 'lg',
+      [layout.small]: !tags && size === 'sm',
```

Comment:
For now, both medium and large sizes support tags, but when we add tag sizes, this restriction will be removed